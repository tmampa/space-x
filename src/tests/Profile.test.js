import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/configureStore';
import MyProfile from '../pages/MyProfile';

describe('Navigation Links Tests', () => {
  test('Check if Component Renders in DOM', () => {
    render(
      <Provider store={store}>
        <MyProfile />
      </Provider>,
    );
    expect(screen.queryByText(/Missions/)).toBeTruthy();
  });
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <MyProfile />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
