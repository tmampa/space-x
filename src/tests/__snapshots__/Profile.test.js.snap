// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation Links Tests renders correctly 1`] = `
<div
  className="profile-page-container"
>
  <figure>
    <figcaption>
      My Missions
    </figcaption>
    <ul
      className="active-list"
    />
  </figure>
  <figure>
    <figcaption>
      My Rockets
    </figcaption>
    <ul
      className="active-list"
    />
  </figure>
</div>
`;
